PR: https://github.com/odoo/odoo/pull/238084

From: 8128a1bfbdd7b2f7cdc8ccbf1ddf78219a809e38
From: jeanschoenlaub
Date: 2025-12-19 08:48:44

Breaking data model changes scores: del:4 + add:0, change matches:
-    virtual_available_at_date = fields.Float(compute='_compute_qty_at_date', digits='Product Unit')
-    scheduled_date = fields.Datetime(compute='_compute_qty_at_date')
-    qty_available_today = fields.Float(compute='_compute_qty_at_date')
-    is_storable = fields.Boolean(related='product_id.is_storable')

Total Changes: 4

[IMP] sale,sale_stock: Add SO line availability widget without stock

On relevant sale order lines, displays a similar widget to what
we have with sale_stock but simplified. The widget shows the
current qty on hand of the SO line product and its forecasted
quantity at the commitment date of the SO.

WHY: When creating a SO, its now easy to view if we can sell right
now (with on hand) and wether we need to create a PO for futur.

Note: because this was a bit rushed for the freeze, I duplicated
the existing widget logic, but `simple_qty_at_date_widget` js and
xml could be inherited in `sale.stock`.

task#5186344

Part-of: odoo/odoo#238084
Related: odoo/upgrade#9081
Signed-off-by: Arnold Moyaux (arm) <arm@odoo.com>

================================= pseudo patch: =================================

--- a/addons/sale_stock/models/sale_order_line.py
+++ b/addons/sale_stock/models/sale_order_line.py
@@ -15,16 +15,12 @@ class SaleOrderLine(models.Model):
     qty_delivered_method = fields.Selection(selection_add=[('stock_move', 'Stock Moves')])
     route_ids = fields.Many2many('stock.route', string='Routes', domain=[('sale_selectable', '=', True)], ondelete='restrict')
     move_ids = fields.One2many('stock.move', 'sale_line_id', string='Stock Moves')
-    virtual_available_at_date = fields.Float(compute='_compute_qty_at_date', digits='Product Unit')
-    scheduled_date = fields.Datetime(compute='_compute_qty_at_date')
     forecast_expected_date = fields.Datetime(compute='_compute_qty_at_date')
     free_qty_today = fields.Float(compute='_compute_qty_at_date', digits='Product Unit')
-    qty_available_today = fields.Float(compute='_compute_qty_at_date')
     warehouse_id = fields.Many2one('stock.warehouse', compute='_compute_warehouse_id', store=True)
     qty_to_deliver = fields.Float(compute='_compute_qty_to_deliver', digits='Product Unit')
     is_mto = fields.Boolean(compute='_compute_is_mto')
     display_qty_widget = fields.Boolean(compute='_compute_qty_to_deliver')
-    is_storable = fields.Boolean(related='product_id.is_storable')
     customer_lead = fields.Float(
         compute='_compute_customer_lead', store=True, readonly=False, precompute=True,
         inverse='_inverse_customer_lead')
